synapse/lib/cell.py

```diff
@@ -1662,36 +1668,46 @@ async def _runSysctlLoop(self):
 
             await self.waitfini(self.SYSCTL_CHECK_FREQ)
 
-    def _getAhaAdmin(self):
-        name = self.conf.get('aha:admin')
-        if name is not None:
-            return name
-
     async def _initAhaRegistry(self):
 
-        ahaurl = self.conf.get('aha:registry')
-        if ahaurl is not None:
+        ahaurls = self.conf.get('aha:registry')
+        if ahaurls is not None:
 
-            info = await s_telepath.addAhaUrl(ahaurl)
-            if self.ahaclient is None:
-                self.ahaclient = await s_telepath.Client.anit(info.get('url'))
-                self.ahaclient._fini_atexit = True
-                self.onfini(self.ahaclient)
+            info = await s_telepath.addAhaUrl(ahaurls)
+            if self.ahaclient is not None:
+                await self.ahaclient.fini()
 
-            async def finiaha():
-                await s_telepath.delAhaUrl(ahaurl)
+            async def onlink(proxy):
+                ahauser = self.conf.get('aha:user', 'root')
+                newurls = await proxy.getAhaUrls(user=ahauser)
+                oldurls = tuple(self.conf.get('aha:registry'))
+                if newurls and newurls != oldurls:
+                    self.modCellConf({'aha:registry': newurls})
```

Comment:
This **could** break existing AHA deployments which use static configuration information ( without provisioning / client side certs ) if are setup in the following fashion:

1. AHA has an aha:name and aha:network setup
2. AHA is listening on a tcp:// listener
3. client cells have an aha:registry configuration set to talk to AHA via TCP.

The client would then get a generated list of ssl:// target URLS which would cause the aha.client reconnects to fail and subsequent service startups to fail.